dbus_service.py

```diff
@@ -693,12 +631,22 @@ def set_dbus_values(self):
                 self._dbusservice[pre + "/Energy/Forward"] = pvyield
                 self._dbusservice["/Ac/Energy/Forward"] = pvyield
 
-            logging.debug("Inverter #%d Power (/Ac/Power): %s",
-                          self.pvinverternumber, power)
-            logging.debug("Inverter #%d Energy (/Ac/Energy/Forward): %s",
-                          self.pvinverternumber, pvyield)
+            logging.debug(f"Inverter #{self.pvinverternumber} Power (/Ac/Power): {power}")
+            logging.debug(f"Inverter #{self.pvinverternumber} Energy (/Ac/Energy/Forward): {pvyield}")
             logging.debug("---")
 
+    def get_ac_inverter_state(self, current):
+        '''return the state of the inverter based on the current value'''
+        try:
+            float_current = float(current)
+        except ValueError:
+            float_current = 0
```

Comment:
Let's discuss, if this is a problem when single data records are missing. In those cases the value should not jump to zero. But I assume, this will not be a problem because in the cases I am thinking of the complete records can not be fetched (timeout, null instead of complete JSON...).

Reply:
Good point! I'm not sure if this could happen. We handle this in fetch_url already, don't we? What I mean is, we rely on fetching the data correctly not only for an inverter. It is quite the same within a PV-inverter.  To avoid problems we should have added an additional check that we got a minimum set of data depending on the class that we suppose to use.

Reply:
Yes, correct. I like the idea of adding some additional simple "health check" after having fetched the JSON. Something like just checking if the json contains some specific entry. Would you like to do that? Then, we definitely have no problem here in this function